Add contributor guidelines & code of conduct update (#679)
typelevel/typelevel.github.com: src/blog/contributing-guidelines-2026.md

{%

>   author: ${foundation}

Reviewer: We could add a TSC author and make it a co-authored post. I would like to somehow give credit to the TSC for your work. (Alternative to co-authoring, could give credit in the post.)

```suggestion
  author: [${tsc}, ${foundation}]
```

  date: "2026-07-25"
  tags: [governance]
%}

# AI Tools, Contributing and Code of Conduct Updates

## New Contributing Guide

> We have published a new [contributing guide](../projects/contributing-guide.md) for Typelevel organization projects and a [start contributing guide](../community/start-contributing.md) for new contributors!

Reviewer: ```suggestion
We have published a new [contributing guide](/projects/contributing-guide.md) for Typelevel organization projects and a [start contributing guide](/community/start-contributing.md) for new contributors!
```

These can be absolute URLs since they are not resources relative to the blog post.

> These guidelines cover similar ground to [the ones](../gsoc/ai.md) we have provided to Google Summer of Code applicants in the past. The new guide is an evergreen document to point any new contributor to for guidance. It covers where to start on an issue, copyright, and how to structure pull requests.

Reviewer: ```suggestion
These guidelines cover similar ground to [the ones](/gsoc/ai.md) we have provided to Google Summer of Code applicants in the past. The new guide is an evergreen document to point any new contributor to for guidance. It covers where to start on an issue, copyright, and how to structure pull requests.
```

## Code of Conduct Update

We have also made a small addition to the [code of conduct](../code-of-conduct/README.md), adding a new behavior to the "Our Community" section:

> - **Using your own words.** We take the time to communicate with our own thoughts and writing.

> This is Typelevel's current restriction on AI-generated content: Community members should not be using these tools to generate content for others to read.

Reviewer 2: English is not my first language, but by my reading this sentence forbids LLM-generated source code contributions. Since the next section explicitly allows these, I think this is not intentional. (Note: source code in a PR _is_ "content for others to read". Namely, for the reviewers. At least, I don't know of a way to review a PR without reading the diff...)

> ## AI Related Policies

Reviewer: ```suggestion
## AI-related Policies
```

> With AI coding tools coming into wide use, it has become very easy for potential contributors to write code without thinking, resulting in contributions that are not actually helpful. To mitigate this, the contributing guide has two guidelines regarding AI-assisted contributions specifically. Individual projects may have additional restrictions in their own contribution policies, but the below is our default expectation:

Reviewer: > write code without thinking, resulting in contributions that are not actually helpful

I wonder if we can rephrase to soften it a bit. Even if it's technically true, we can consider the contributor's intent which was hopefully to help :)

Author: Perhaps I can focus on the effort asymmetry?

```suggestion
With AI coding tools coming into wide use, it has become very easy for potential contributors to write code far faster than maintainers can review. This doesn't help contributors get involved in projects or let maintainers get any more work done. At worst, maintainers get bogged down addressing pull requests that have never been looked at by a person before. To mitigate this, the contributing guide has two guidelines regarding AI-assisted contributions specifically. Individual projects may have additional restrictions in their own contribution policies, but the below is our default expectation:
```

With agentic coding tools coming into wide use, it has become very easy for potential contributors to write code without thinking, resulting in contributions that are not actually helpful. To mitigate this, the contributing guide has two guidelines regarding AI-assisted contributions specifically. Individual projects may have additional restrictions in their own contribution policies, but the below is our default expectation:

* We **are** allowing AI-assisted coding contributions. We expect contributors to disclose when they have used such tools and be responsible for their contributions regardless of how they are coded.

> * We are **not** allowing AI generated pull requests, issues descriptions, or other prose. We expect that contributors understand their work well enough to explain it themselves.

Reviewer: ```suggestion
* We are **not** allowing AI-generated pull requests, issues descriptions, or other prose. We expect that contributors understand their work well enough to explain it themselves.
```

> * We are **not** allowing AI generated pull requests, issues descriptions, or other prose. We expect that contributors understand their work well enough to explain it themselves.

Reviewer 2: If LLM-generated PR _descriptions_ are disallowed, while LLM-generated source code changes in a PR are allowed, maybe this sentence could make this clearer.

Pull request descriptions, pull request comments, discussions, Markdown documents, Discord messages, and such are all examples of something very important: human beings communicating with other human beings. Generating a long and wordy description of something which can be more concisely and organically expressed serves no one's best interests. Rubber stamping something done even more autonomously by an AI in your name is even less useful.

> Maintainers have very, very limited time that they are able to devote to their projects. As a contributor, please respect the time they are giving you by giving them your attention when you interact with Typelevel projects, rather than allowing an agentic system to step in the way.

Reviewer 2: I like the 2nd sentence, but not the 1st one. Is the "very limited time" really the thing to point out here? It is certainly true. But I think the "respect the time of your fellow human beings" (paraphrased) from the 2nd sentence is more important.

### Use of Translation Tools

> We do understand that LLMs are excellent tools for bridging language barriers, and as always, we strive for the most inclusive environment possible on our projects and in our community spaces. To that end, we do make an exception for the use of AI tooling as a linguistic aid when English is not your primary language. Specifically, when you are authoring your own words in your own language and then leveraging a model to translate those words to English prior to posting in a Typelevel forum, this is considered acceptable and even encouraged behavior.

Reviewer 3: Small suggested tweak for clarity:

```suggestion
We do understand that LLMs are excellent tools for bridging language barriers, and as always, we strive for the most inclusive environment possible on our projects and in our community spaces. To that end, we allow the use of AI tooling as a linguistic aid when English is not your primary language. Specifically, when you are authoring your own words in your own language and then leveraging a model to translate those words to English prior to posting in a Typelevel forum, this is considered acceptable and even encouraged behavior.
```
